### PR TITLE
Topic recognition

### PR DESCRIPTION
**Pull request by LargePenguin034**
Improved Unet model that is used to map a mask to skin Lessions in the ISIC dataset

Contains

dataset.py - Function for data loading
modules.py - Improved Unet model
predict.py - Used to use model after training
train.py - Used to train model

Aaron Reynolds
Student ID: 47479049

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
This is an initial inspection, no action is required at this point.

- Files are a bit chaos, but containing all necessary ones. Trained model is included in the repository, which is not allowed as per the spec.
- Algorithm solves the problem appropriately. Average Dice Coefficient in testing dataset: 0.88, but no evidence.
- Correctly using PyTorch to construct the Improved U-Net models and functions.
- Good coding design. Well handwriting code.
- Sufficient docstring and comment in the code.
- Easy difficulty problem, no higher difficulty marks.

- Wrong pull request branch, from local:topic-recognition branch to shakes76:main branch.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting minimal -1
No Data leakage
Difficulty: Easy -10
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe OK, missing usage and other info -2
Model/technical explanation, could be better -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in InCorrect Branch) -2
Feedback required, remove model files and correct branch and remove gitignore files -2
Request Description OK

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.
